pkg/controller/ec2/vpcpeeringconnection/setup.go

```diff
@@ -0,0 +1,189 @@
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube   client.Client
+	client svcsdkapi.EC2API
+}
+
+func preObserve(ctx context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsInput) error {
+	filterName := "tag:Name"
+	externalName := meta.GetExternalName(cr)
+	filterValue := []*string{&externalName}
+	filter := svcsdk.Filter{
+		Name:   &filterName,
+		Values: filterValue,
+	}
+	filters := []*svcsdk.Filter{&filter}
+	obj.SetFilters(filters)
+	return nil
+}
+
+func (e *custom) postObserve(_ context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	connectionCounter := -1
+	for _, v := range obj.VpcPeeringConnections {
+		connectionCounter++
+		for _, tag := range v.Tags {
+			if *tag.Key == "Name" {
+				if *tag.Value == cr.ObjectMeta.Labels["peering-connection-id"] {
+					switch aws.StringValue(v.Status.Code) {
+					case string(svcapitypes.VPCPeeringConnectionStateReasonCode_pending_acceptance):
+						cr.SetConditions(xpv1.Creating())
+					case string(svcapitypes.VPCPeeringConnectionStateReasonCode_deleted):
+						cr.SetConditions(xpv1.Unavailable())
+						return managed.ExternalObservation{}, nil
+					case string(svcapitypes.VPCPeeringConnectionStateReasonCode_active):
+						cr.SetConditions(xpv1.Available())
+					}
+				}
+			}
+		}
+	}
+	cr.SetConditions(xpv1.Available())
+	return obs, nil
+}
+
+func (e *custom) isUpToDate(cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsOutput) (bool, error) {
+	connectionCounter := -1
+
+	for _, v := range obj.VpcPeeringConnections {
+		connectionCounter++
+
+		for _, tag := range v.Tags {
+			if *tag.Key == "Name" {
+				if *tag.Value == cr.ObjectMeta.Name {
+					switch *v.Status.Code {
+					case "active":
+						return true, nil
+					case "deleted":
+						return false, nil
+					}
+
+					if *v.Status.Code == "pending-acceptance" && cr.Spec.ForProvider.AcceptRequest {
+						// if acceptRequest is true, we automatically accept the request on AWS
+						req := svcsdk.AcceptVpcPeeringConnectionInput{
+							VpcPeeringConnectionId: awsclients.String(*v.VpcPeeringConnectionId),
+						}
+
+						request, _ := e.client.AcceptVpcPeeringConnectionRequest(&req)
+						err := request.Send()
+						if err != nil {
+							return false, err
+						}
+						return true, nil
+					}
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+func preCreate(ctx context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.CreateVpcPeeringConnectionInput) error {
+	// set external name as tag on the vpc peering connection
+	resType := "vpc-peering-connection"
+	key := "Name"
```

Comment:
```suggestion
	key := "crossplane-external-name"
```
so that it's more clear who set this. Also, we can have this as a `const` and use it in all functions.

Reply:
I was using the Name tag because this will also be the display name in aws console